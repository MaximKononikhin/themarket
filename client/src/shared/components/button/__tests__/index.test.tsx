import React from "react";

import { render } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

import Button from "@shared/components/button";

const TEST_CONTENT = "test";

describe("Button component", () => {
    it("should render component", () => {
        const { getByText } = render(<Button>{TEST_CONTENT}</Button>);

        const container = getByText(TEST_CONTENT);

        expect(container).toBeInTheDocument();
        expect(container).toMatchSnapshot();
    });

    it("should call handleClick", async () => {
        const handleClick = jest.fn();

        const { getByText } = render(
            <Button onClick={handleClick}>{TEST_CONTENT}</Button>
        );

        const container = getByText(TEST_CONTENT);

        await userEvent.click(container);

        expect(handleClick).toHaveBeenCalledTimes(1);
    });
});
