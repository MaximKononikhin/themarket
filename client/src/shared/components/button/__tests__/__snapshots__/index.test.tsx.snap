// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component should render component 1`] = `
<button
  class="button"
>
  test
</button>
`;
